### PR TITLE
Make a better "Mod download confirmation dialog"

### LINKED ISSUE #263: mod install confirmation window goes off screen
### Operating System

Linux (KDE)

### Description of bug

when you select a lot of mods to install, the confirmation window is too long to be able to reach the ok button
oof

### Steps to reproduce

select a large amount of mods to install 
you won't be able to reach the button 

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by flowln**
Fixes #263. This replaces the previous UI, that was made with a previously existing dialog, with a more refined one, that allows for scrolling, makes it easier to add more information later on, would allow for removing mods on the confirmation dialog itself (feature idea ;) ), and generally looks nicer in my opinion :D

### CODE REVIEW

**Comment by Scrumplex** [comment]
Screenshot if you are too lazy to compile the changes: Screenshot of dialog

**Review by Scrumplex — APPROVED** [review]
Looks good, code-wise and UI-wise :D

**Comment by Scrumplex** [comment]
Not sure if we want to cram this into 1.1.1 or 1.2.0

**Comment by txtsd** [comment]
> Not sure if we want to cram this into 1.1.1 or 1.2.0

It's a bugfix technically, so 1.1.1 makes sense.

**Review by txtsd — APPROVED** [review]

**Comment by Scrumplex** [comment]
yeah fine by me :D

File: launcher/ui/dialogs/ReviewMessageBox.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    return new ReviewMessageBox(parent, title, icon);
+}
+
+void ReviewMessageBox::appendMod(const QString& name, const QString& filename)
+{
+    auto itemTop = new QTreeWidgetItem(ui->modTreeWidget);
+    itemTop->setText(0, name);
+
+    auto filenameItem = new QTreeWidgetItem(itemTop);
+    filenameItem->setText(0, QString("Filename: %1").arg(filename));
```
Comment (flowln) [inline launcher/ui/dialogs/ReviewMessageBox.cpp]: wait, does this need tr()? Or does QString already do this for us?

Comment (flowln) [reply launcher/ui/dialogs/ReviewMessageBox.cpp]: nvm seems like QString does this already

Comment (Scrumplex) [reply launcher/ui/dialogs/ReviewMessageBox.cpp]: hmm not sure to be honest. Can't find any info about this

Comment (flowln) [reply launcher/ui/dialogs/ReviewMessageBox.cpp]: I found this: https://doc.qt.io/qt-5/i18n-source-translation.html#using-numbered-arguments
I'll use it from now on, it's much less confusing xD

**Review by ZekeZDev — APPROVED** [review]

**Pull request merged**